sw/device/tests/crypto/ecdsa_p256_verify_functest.c

```diff
@@ -0,0 +1,88 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/crypto/drivers/hmac.h"
+#include "sw/device/lib/crypto/drivers/otbn.h"
+#include "sw/device/lib/crypto/ecdsa_p256/ecdsa_p256.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+#include "sw/device/tests/crypto/ecdsa_p256_verify_testvectors.h"
+
+hmac_error_t compute_digest(size_t msg_len, const uint8_t *msg,
```

Comment:
Can we factor this out as a general utility? E.g. something like
```c
void compute_digest(const void *msg, size_t msg_len, void *out);
```
(And have it fail the test if digest computation fails.)

We can punt to another CL just to get this one in now.

Reply:
I'm planning a few adjustments in general now that we have the API more figured out -- I think I'll probably remove the custom `ecdsa_p256_message_digest_t` struct and have it just take a sha256 digest type which would be defined in the hash functions section of the library, and then simply call a function from there to compute the digest. So -- coming soon to a cryptolib near you!